nucypher/characters/chaotic.py

```diff
@@ -95,3 +114,355 @@ def status():
+        self._distribution_task.stop()
+        return True
+
+    def __calculate_disbursement(self, recipient) -> int:
+        """Calculate the next reward for a recipient once the are selected for distribution"""
+
+        # Initial Reward - sets the future rates
+        if recipient.last_disbursement_time is None:
+            amount = self.INITIAL_DISBURSEMENT
+
+        # Cap reached, We'll continue to leak the minimum disbursement
+        elif int(recipient.total_received) >= self.MAXIMUM_DISBURSEMENT:
+            amount = self.MINIMUM_DISBURSEMENT
+
+        # Calculate the next disbursement
+        else:
+            amount = math.ceil(int(recipient.last_disbursement_amount) * self.MULTIPLIER)
+            if amount < self.MINIMUM_DISBURSEMENT:
+                amount = self.MINIMUM_DISBURSEMENT
+
+        return int(amount)
+
+    def __transfer(self, disbursement: int, recipient_address: str) -> str:
+        """Perform a single token transfer transaction from one account to another."""
+
+        self.__disbursement += 1
+        txhash = self.token_agent.transfer(amount=disbursement,
+                                           target_address=recipient_address,
+                                           sender_address=self.checksum_public_address)
+
+        self.log.info(f"Disbursement #{self.__disbursement} OK | {txhash.hex()[-6:]} | "
+                      f"({str(NU(disbursement, 'NuNit'))}) -> {recipient_address}")
+        return txhash
+
+    def airdrop_tokens(self):
+        """
+        Calculate airdrop eligibility via faucet registration
+        and transfer tokens to selected recipients.
+        """
+
+        with ThreadedSession(self.db_engine) as session:
+            population = session.query(self.Recipient).count()
+
+        message = f"{population} registered faucet recipients; " \
+                  f"Distributed {str(NU(self.__distributed, 'NuNit'))} since {self.start_time.slang_time()}."
+        self.log.debug(message)
+        if population is 0:
+            return  # Abort - no recipients are registered.
+
+        # For filtration
+        since = datetime.now() - timedelta(hours=self.DISBURSEMENT_INTERVAL)
+
+        datetime_filter = or_(self.Recipient.last_disbursement_time <= since,
+                              self.Recipient.last_disbursement_time == None)
+
+        with ThreadedSession(self.db_engine) as session:
+            candidates = session.query(self.Recipient).filter(datetime_filter).all()
+            if not candidates:
+                self.log.info("No eligible recipients this round.")
+                return
+
+        # Discard invalid addresses, in-depth
+        invalid_addresses = list()
+
+        def siphon_invalid_entries(candidate):
+            address_is_valid = eth_utils.is_checksum_address(candidate.address)
+            if address_is_valid:
+                invalid_addresses.append(candidate.address)
```

Comment:
Sounds like you understanding is correct.  There is a missing `not` inside this helper function.  

I'm using the `siphon_invalid_entries` to return `True` is a given element is a valid checksum address.  Used as the filter function, it will remove invalid addresses, and track the invalid ones in a list, resulting in `candidates`.

Reply:
Sure - but does that mean that `invalid_addresses` isn't doing anything?

Reply:
Yes.  The resulting list `invalid_addresses` is  pretty much just used to generate a log message.

Reply:
Do you think we want to prune the database of bad entries here, or what are you thinking?

Reply:
@jMyles - Here's a stub to follow-up: <https://github.com/nucypher/nucypher/pull/812/commits/d71d76beb35e94790fb5ccfb244051d41b128a62#diff-9f8c085f8e0727aed7eed6c267d99904R410>